wheel/pep425tags.py

```diff
@@ -105,17 +109,85 @@ def get_abi_tag():
     return abi
 
 
-def get_platform():
+def calculate_macosx_platform_tag(archive_root, platform_tag):
+    """
+    Calculate proper macosx platform tag basing on files which are included to wheel
+    """
+    prefix, base_version, suffix = platform_tag.split('-')
+    base_version = tuple([int(x) for x in base_version.split(".")])
+    if len(base_version) == 2:
+        base_version = base_version + (0,)
+
+    assert len(base_version) == 3
+    if "MACOSX_DEPLOYMENT_TARGET" in os.environ:
+        deploy_target = tuple([int(x) for x in os.environ[
+            "MACOSX_DEPLOYMENT_TARGET"].split(".")])
+        if len(deploy_target) == 2:
+            deploy_target = deploy_target + (0,)
+        if deploy_target < base_version:
+            sys.stderr.write(
+                "[WARNING] MACOSX_DEPLOYMENT_TARGET is set "
+                "to lower value than your python is compiled\n"
```

Comment:
It mean that you can use higher version of `MACOSX_DEPLOYMENT_TARGET` than is set in environment variable because used python interpreter is compiled against newer system api. So you cannot create wheel for this system version (chosen by environment variable) using this setup.

Reply:
OK, right. So maybe we should tell the user?

What about something like this, then, giving a bit more information?

`"[WARNING] MACOSX_DEPLOYMENT_TARGET is set to a lower value than the version on which the Python interpreter was compiled, and will be ignored.\n"`

Or even, to be really helpful:
`"[WARNING] MACOSX_DEPLOYMENT_TARGET is set to a lower value ({}) than the version on which the Python interpreter was compiled ({}), and will be ignored.\n".format('.'.join(str(x) for x in deploy_target), '.'.join(str(x) for x in base_version))`

Reply:
Current python gives additional information (python compiled against 10.14):

```bash
$ MACOSX_DEPLOYMENT_TARGET="10.12" python setup.py bdist_wheel
...
MACOSX_DEPLOYMENT_TARGET mismatch: now "10.12" but "10.14" during configure
``` 
So I'm not sure if this text is need and in which form.

Reply:
Well, if you ask me, the current message makes me wonder what impact this has on the build. Maybe you know the whole `MACOSX_DEPLOYMENT_TARGET` thing too well by now, and it's obvious to you? :-D